File: source/clibCore/addonCollection_core/clickEmail.vb

Option Explicit On
Option Strict On

Imports Contensive.BaseClasses
Imports Contensive.Core
Imports Contensive.Core.Models.Entity
Imports Contensive.Core.Controllers
Imports Contensive.Core.Controllers.genericController
'
Namespace Contensive.Addons.Core
    Public Class clickEmailClass
        Inherits Contensive.BaseClasses.AddonBaseClass
        '
        '====================================================================================================
        ''' <summary>
        ''' getFieldEditorPreference remote method
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <returns></returns>
        Public Overrides Function execute(cp As Contensive.BaseClasses.CPBaseClass) As Object
            Dim result As String = ""
            Try
                Dim processor As CPClass = DirectCast(cp, CPClass)
                Dim cpCore As coreClass = processor.core
                '
                ' -- Email click detected
                Dim emailDrop As emailDropModel = emailDropModel.create(cpCore, cpCore.docProperties.getInteger(rnEmailClickFlag))
                If (emailDrop IsNot Nothing) Then
                    Dim recipient As personModel = personModel.create(cpCore, cpCore.docProperties.getInteger(rnEmailMemberID), New List(Of String))
                    If (recipient IsNot Nothing) Then
                        Dim log As New Models.Entity.emailLogModel() With {
                            .name = "User " & recipient.name & " clicked link from email drop " & emailDrop.name & " at " & cpCore.profileStartTime.ToString(),
                            .EmailDropID = emailDrop.id,
                            .MemberID = recipient.id,
                            .LogType = EmailLogTypeOpen
                        }
                    End If
                End If
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
            End Try
            Return result
        End Function
    End Class
End Namespace

File: source/core/addonCollection_core/clickEmail.cs

using System;
using System.Reflection;
using System.Xml;
using System.Diagnostics;
using System.Linq;
using System.Text.RegularExpressions;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using Contensive.Core;
using Contensive.Core.Models.Entity;
using Contensive.Core.Controllers;
using static Contensive.Core.Controllers.genericController;
using static Contensive.Core.constants;
//
namespace Contensive.Addons.Core {
    public class clickEmailClass : Contensive.BaseClasses.AddonBaseClass {
        //
        //====================================================================================================
        /// <summary>
        /// getFieldEditorPreference remote method
        /// </summary>
        /// <param name="cp"></param>
        /// <returns></returns>
        public override object Execute(Contensive.BaseClasses.CPBaseClass cp) {
            string result = "";
            try {
                CPClass processor = (CPClass)cp;
                coreClass cpCore = processor.core;
                //
                // -- Email click detected
                emailDropModel emailDrop = emailDropModel.create(cpCore, cpCore.docProperties.getInteger(rnEmailClickFlag));
                if (emailDrop != null) {
                    personModel recipient = personModel.create(cpCore, cpCore.docProperties.getInteger(rnEmailMemberID));
                    if (recipient != null) {
                        emailLogModel log = new emailLogModel() {
                            name = "User " + recipient.name + " clicked link from email drop " + emailDrop.name + " at " + cpCore.doc.profileStartTime.ToString(),
                            EmailDropID = emailDrop.id,
                            MemberID = recipient.id,
                            LogType = EmailLogTypeOpen
                        };
                    }
                }
            } catch (Exception ex) {
                cp.Site.ErrorReport(ex);
            }
            return result;
        }
    }
}
